### PR TITLE
Add IDENTITY_IF_ABSENT label

### PR DESCRIPTION
**Pull request by oldergod**
I don't think we need the ONE_OF value for oneOfs. Right now, they have the `OPTIONAL` label (which is a lie) but they are treated as optional fields in both proto2 and 3 so I am leaving them as is. Let me know if you wanna set their label to `ONE_OF`.

### CODE REVIEW

File: wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/WireField.kt [hunk 1]
```diff
@@ -66,7 +66,14 @@ annotation class WireField(
...
     ONE_OF,
     /** Implies [REPEATED]. */
-    PACKED;
+    PACKED,
+    /**
+     * Special label to defines proto3 fields which should not be emitted if their value is equal
+     * to their type's respective identity value. E.g.: a field of type `int32` will not get emitted
+     * if its value is `0`.
+     */
+    IDENTITY_IF_ABSENT,
```
Comment (JakeWharton) [inline wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/WireField.kt:75]: The doc describes "absent if identity" which is a property of encoding and not "identity if absent" which would be a property of decoding. Should this be named `ABSENT_IF_IDENTITY`?

Comment (oldergod) [reply wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/WireField.kt:75]: This actually go both ways, right. I think all those labels can be read for both encoding and decoding. Maybe just `IDENTITY`?

Comment (JakeWharton) [reply wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/WireField.kt:75]: Ah, I see.

`IGNORE_IDENTITY`? `OMIT_IDENTITY`?

File: wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java [hunk 2]
```diff
@@ -1151,12 +1151,28 @@ private AnnotationSpec wireFieldAnnotation(NameAllocator nameAllocator, Field fi
...
-    if (!field.isOptional()) {
-      if (field.isPacked()) {
-        result.addMember("label", "$T.PACKED", WireField.Label.class);
-      } else if (field.getLabel() != null) {
-        result.addMember("label", "$T.$L", WireField.Label.class, field.getLabel());
-      }
+    WireField.Label wireFieldLabel;
+    //noinspection ConstantConditions
+    switch (field.getEncodeMode()) {
+      case THROW_IF_ABSENT:
```
Comment (swankjesse) [inline wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java:1157]: should we rename `THROW_IF_ABSENT` to `REQUIRED` ? Fewer names is better

File: wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/WireField.kt (hunk 1 above)
Comment (swankjesse) [reply wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/WireField.kt:75]: I like `OMIT_IDENTITY`

File: wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java (hunk 2 above)
Comment (oldergod) [reply wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java]: fixed

**Comment by oldergod** [comment]
I went with `OMIT_IDENTITY`

**Pull request merged**